### ISSUE TITLE
Disable LanguageTool for Markdown

### ISSUE THREAD

**Issue by oblitum** [dense-analysis/ale#3379]
LanguageTool [_is not_](https://github.com/languagetool-org/languagetool/issues/445#issuecomment-236883920) proper to handle Markdown content. The presence of the tool in the command line will cause ALE by default to render Markdown files full of not relevant warning, simply because LanguageTool doesn't understand the syntax, it can only proofread plain text.

I don't use AsciiDoc, but it may be in the same boat, or worse.

I recommend that ALE changes this and leave LanguageTools for plain text files only.

To use LanguageTool for Markdown, it's best to employ a wrapper, as attempted at #3376 with TeXtidote.

**Comment by hsanson** [comment]
There is people that do use languagetool for Markdown and Asciidoc:

 - https://github.com/dense-analysis/ale/pull/3334
 - https://github.com/dense-analysis/ale/pull/3229
 - https://github.com/dense-analysis/ale/issues/3220
 - https://github.com/dense-analysis/ale/pull/2155

Rather that remove support for LanguateTool for asciidoc and markdown that seems to be useful for some I would recommend to g:ale_linters variable to disable it for these file types.

**Comment by oblitum** [comment]
Changed back title to disable. FWIW I think most of those user would be much more happy by using a proper wrapper instead, the LanguageTool experience on Markdown is simply bad.

**Comment by hsanson** [comment]
Add this to your configuration for the desired result:

```
let g:ale_linters_ignore = {
  \   'markdown': ['languagetool'],
  \   'asciidoc': ['languagetool']
  \}
```

**Comment by oblitum** [comment]
@hsanson it's not personal issue, I know that conf in particular already. The issue is about disabling it by default. As stated already, languagetool is not a tool for parsing Markdown, according to the authors themselves.

**Comment by JensHeinrich** [comment]
Yes, it is not but this could be considered a breaking change I think.
My suggestion would be to disable languagetool as a direct linter, if the textidote wrapper is available to get the best of both worlds.

**Comment by oblitum** [comment]
@JensHeinrich could be a breaking change for those that consider it more useful than problematic, but also could be a relief for those that consider their markdown are bing gratuitously filled with false positives. This was overlooked when added, it was a mistake (I was at the issue thread that added languagetool, and thought it was great).

**Comment by karlb** [comment]
So far I only got the "Possible typo: you repeated a whitespace" warning and am quite happy with the languagetool integration otherwise. Maybe we can ignore that rule for markdown by default?
